src/applications/personalization/profile-2/tests/e2e/profile.personal-and-contact-information-modals.cypress.spec.js

```diff
@@ -0,0 +1,176 @@
+import { PROFILE_PATHS } from '../../constants';
+
+import mockUser from '../fixtures/users/user-36.json';
+import mockFeatureToggles from '../fixtures/feature-toggles.json';
+
+const setup = (mobile = false) => {
+  window.localStorage.setItem(
+    'DISMISSED_ANNOUNCEMENTS',
+    JSON.stringify(['single-sign-on-intro']),
+  );
+
+  if (mobile) {
+    cy.viewport('iphone-4');
+  }
+
+  cy.login(mockUser);
+  cy.route('GET', '/v0/feature_toggles*', mockFeatureToggles);
+  cy.visit(PROFILE_PATHS.PROFILE_ROOT);
+
+  // should show a loading indicator
+  cy.findByRole('progressbar').should('exist');
+  cy.findByText(/loading your information/i).should('exist');
+
+  // and then the loading indicator should be removed
+  cy.findByText(/loading your information/i).should('not.exist');
+  cy.findByRole('progressbar').should('not.exist');
+};
+
+const checkModals = options => {
+  const {
+    editButtonId,
+    otherSectionEditButtonId,
+    editLineId,
+    sectionName,
+  } = options;
+
+  // Open edit view
+  cy.get(`#${editButtonId}-edit-link`).click({
+    force: true,
+  });
+  // Make an edit
+  cy.get(`#${editLineId}`)
+    .click()
+    .type('test');
+  // Click on a different section to edit
+  cy.get(`#${otherSectionEditButtonId}-edit-link`).click({
+    force: true,
+  });
+
+  // Modal appears
+  cy.get('.va-modal').within(() => {
+    cy.contains(`You’re currently editing your ${sectionName}`).should('exist');
+    cy.findByRole('button', { name: /OK/i }).click({
+      force: true,
+    });
+  });
+
+  // Click on cancel in the current section
+  cy.findByRole('button', { name: /Cancel/i }).click({
+    force: true,
+  });
+
+  // Confirmation modal appears, confirm cancel
+  cy.get('.va-modal').within(() => {
+    cy.contains(`You haven’t finished editing your ${sectionName}.`).should(
+      'exist',
+    );
+    cy.findByRole('button', { name: /Cancel/i }).click({
+      force: true,
+    });
+  });
+};
+
+describe('Modals on the personal information and content page', () => {
+  describe('on Desktop', () => {
+    beforeEach(() => {
+      setup();
```

Comment:
Can we do the `setup()` just once, before all of the test cases? I think it would make the tests run a lot faster.

Reply:
I looked at this and if we make all of the `checkModals()` calls in a single `it()` things run more than twice as fast. Huge speed win. Just need to add comments to take the place of the helpful description strings you pass as the first arg to `it()`

Reply:
I made the updates!

I think you are right in that speed is more important over having more broken up tests that are possibly easier to debug when they fail (when having nested `it`s it is very easy to see exactly test case fails, which is why I had nested them initially).

Reply:
Cypress has a best practice item on this topic; it wasn't my idea to write a bunch of assertions in a single big `it()` :) https://docs.cypress.io/guides/references/best-practices.html#Creating-“tiny”-tests-with-a-single-assertion